src/hausdorff.jl

```diff
@@ -82,35 +82,44 @@ See also: [`hausdorff`](@ref)
 const ModifiedHausdorff = GenericHausdorff{MeanReduction, MaxReduction}
 ModifiedHausdorff() = ModifiedHausdorff(MeanReduction(), MaxReduction())
 
-# convert binary image to a point set format
-function img2pset(img::AbstractArray{T}) where T<:Union{Gray{Bool}, Bool}
-    inds = findall(x->x==true, img)
-    [inds[j][i] for i=1:ndims(img), j=1:length(inds)]
+# convert binary image to its distance transform
+const hausdorff_transform(img::AbstractArray{Bool}) = img |> feature_transform |> distance_transform
+
+const hausdorff_transform(img::GenericGrayImage) = hausdorff_transform(of_eltype(Bool, img))
+
+function evaluate_hausdorff(d::GenericHausdorff,
+                            imgA::AbstractArray{Bool},
+                            imgB::AbstractArray{Bool},
+                            dtA::AbstractArray,
+                            dtB::AbstractArray)
+    # trivial cases
+    imgA == imgB && return 0.
+    (isempty(imgA) || isempty(imgB)) && return Inf
+
+    # dtA and dtB contain the distance from each pixel
+    # to the nearest active pixel in imgA and imgB, respectively.
+    # We only care about the distances from imgA to imgB and vice
+    # versa, so we'll pull those out by using those images as logical
+    # masks into the distance arrays.
+    dAB = _reduce(d.inner_op, dtB[imgA])
```

Comment:
Will the memory allocation be an issue? `dtB[imgA]` versus `view(dtB, imgA)`

Reply:
That is a good point, can you adapt to use `view` instead @jsenn?

Reply:
Didn't know about `view`, thanks for the comment!